src/ui/components/SecondaryToolbox/index.tsx

```diff
@@ -177,9 +177,15 @@ function SecondaryToolbox() {
         <Panel isActive={selectedPanel === "redux-devtools"}>
           <ReduxDevToolsPanel />
         </Panel>
-        <Panel isActive={toolboxLayout !== "ide" && selectedPanel === "debugger"}>
-          <EditorPane />
-        </Panel>
+        {/* _
+          Don't_ render an `<Offscreen>` around the editor.
+          It appears that this causes React to unmount and remount
+          the class components, and we end up with a duplicate
+          CodeMirror instance or similar, and the entire editor freezes
+          and becomes non-interactive. If we skip that, toggling the "Sources" 
+          tab works just fine as expected.
+        */}
+        {toolboxLayout !== "ide" && selectedPanel === "debugger" && <EditorPane />}
```

Comment:
> It appears that this causes React to unmount and remount the class components

Just to clarify, this is kind of what Offscreen is meant to do. Rather than throwaway and recreate, React just holds onto the unmounted instance and remounts it. If a component isn't resilient to this, it will cause problems. That was the motivation behind adding strict effects mode.

I'm guessing the `EditorPane` component (or something inside of it– maybe even Code Mirror itself) isn't resilient to that.

Reply:
Looking at the `Editor` code, it certainly _tries_ to clean up the actual CM instance in `componentWillUnmount`.  Maybe there's something else going on too.